Source/ARTRealtimeChannel.m

```diff
@@ -164,6 +164,59 @@ - (void)internalPostMessages:(id)data callback:(void (^)(ARTErrorInfo *__art_nul
 } ART_TRY_OR_MOVE_TO_FAILED_END
 }
 
+- (void)sync {
+    [self sync:nil];
+}
+
+- (void)sync:(void (^)(ARTErrorInfo *__art_nullable error))callback {
+    if (callback) {
+        void (^userCallback)(ARTErrorInfo *__art_nullable error) = callback;
+        callback = ^(ARTErrorInfo *__art_nullable error) {
+            ART_EXITING_ABLY_CODE(self->_realtime.rest);
+            dispatch_async(self->_userQueue, ^{
+                userCallback(error);
+            });
+        };
+    }
+
+ART_TRY_OR_MOVE_TO_FAILED_START(_realtime) {
+    switch (self.state) {
+        case ARTRealtimeChannelInitialized:
+        case ARTRealtimeChannelDetaching:
+        case ARTRealtimeChannelDetached: {
+            ARTErrorInfo *error = [ARTErrorInfo createWithCode:40000 message:@"unable to sync to channel; not attached"];
+            [self.logger logWithError:error];
+            if (callback) callback(error);
+            return;
+        }
+        default:
+            break;
+    }
+
+    [self.logger verbose:__FILE__ line:__LINE__ message:@"R:%p C:%p (%@) requesting a sync operation", _realtime, self, self.name];
+
+    ARTProtocolMessage * msg = [[ARTProtocolMessage alloc] init];
+    msg.action = ARTProtocolMessageSync;
+    msg.channel = self.name;
+
+    if (self.presenceMap.syncMsgSerial) {
+        msg.channelSerial = self.presenceMap.syncChannelSerial;
+        msg.msgSerial = [NSNumber numberWithLongLong:self.presenceMap.syncMsgSerial];
+    }
```

Comment:
I don't understand this. What is a syncMsgSerial? It looks like it's being set from a SYNC msgSerial, but that isn't a thing (no incoming messages ever have msgSerials, and SYNCs don't have them in either direction). Nothing in RTP3 tells you to do this, it only asks for the channelSerial. 

Also, AFAICS syncChannelSerial and syncMsgSerial are never unset in endSync, so any sync() call will result in the lib trying to resume an actually-successfully-completed sync, no?

Reply:
@SimonWoolf Thanks for pointing this out. Fixed dc84181.

Reply:
And the second issue (second paragraph of my comment)? `presenceMap.syncChannelSerial` is only ever set, never unset, and you're setting it on the sync unconditionally, requesting a resume of a sync that actually completed successfully?

Reply:
@SimonWoolf Right, fixed 0d7cdc0. Set `syncChannelSerial` to nil when the sync ends.